{
  try {
    Record empty=new Record();
    empty.write(this.out);
    empty.read(this.in);
    Assert.assertTrue("Deserialized Empty record is not another empty record.",empty.getNumFields() == 0);
    empty=new Record();
    empty.write(this.out);
    empty=new Record();
    empty.read(this.in);
    Assert.assertTrue("Deserialized Empty record is not another empty record.",empty.getNumFields() == 0);
  }
 catch (  Throwable t) {
    Assert.fail("Test failed due to an exception: " + t.getMessage());
  }
}
